infra/enrolment-records/repository/src/main/java/com/simprints/infra/enrolment/records/repository/commcare/CommCareEventSyncTask.kt

```diff
@@ -0,0 +1,145 @@
+package com.simprints.infra.enrolment.records.repository.commcare
+
+import android.content.Context
+import androidx.core.net.toUri
+import com.simprints.core.DispatcherBG
+import com.simprints.infra.enrolment.records.repository.commcare.domain.CommCareCase
+import com.simprints.infra.logging.Simber
+import dagger.hilt.android.qualifiers.ApplicationContext
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.withContext
+import javax.inject.Inject
+
+/**
+ * Task for syncing CommCare case data and tracking case metadata.
+ * Queries CommCare's content provider to extract case information and maintains
+ * sync tracking records with case IDs, subject IDs, and last modified timestamps.
+ */
+class CommCareEventSyncTask @Inject constructor(
```

Comment:
Issue description says to save the list of cases synced from CommCare when the existing CommCareEventSyncTask runs, not to implement another class with the same name.